internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumer.java

```diff
@@ -0,0 +1,62 @@
+package com.linkedin.venice.client.consumer;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+
+/**
+ * Venice change capture consumer to provide
+ * @param <K>
+ * @param <V>
+ */
+public interface VeniceChangelogConsumer<K, V> {
+  /**
+   * Subscribe a set of partitions to this VeniceChangelogConsumer. The VeniceChangelogConsumer
+   * should try to consume messages from all partitions that are subscribed to it.
+   *
+   * @return a future which completes when the partitions are ready to be consumed data
+   * @param partitions the set of partition to subscribe and consume
+   * @throws a VeniceException if subscribe operation failed for any of the partitions
+   */
+  CompletableFuture<Void> subscribe(Set<Integer> partitions);
+
+  /**
+   * Subscribe all partitions belonging to a specific store.
+   *
+   * @return a future which completes when all partitions are ready to be consumed data
+   * @throws a VeniceException if subscribe operation failed for any of the partitions
+   */
+  CompletableFuture<Void> subscribeAll();
+
+  /**
+   * Obtain the underlying store name.
+   *
+   * @return store name
+   */
+  String getStoreName();
```

Comment:
I guess maybe I have some reservation as adding this might imply that we can't ever have an implementation of this consumer interface which consumes records from multiple stores (if I think about the Kafka Consumer interface you can have one client to many topics).

Reply:
Yeah, I borrow this from davinci interface. It is also related to `subscribe(Set<Integer> partitions)` interface. If we think we really want to support multiple stores, we might need to change all of these interfaces.

Reply:
I'm not sure I understand why, but supporting that isn't really in scope so lets put it aside.  Question I'd like to discuss is do we need this method in the interface?

Reply:
I guess we do not need at this moment.